data/campaigns/Eastern_Invasion/utils/macros.cfg

```diff
@@ -1,87 +1,395 @@
-                set_type=arcane
-            [/effect]
-            [effect]
-                apply_to=attack
-                range=ranged
-                set_type=arcane
-            [/effect]
-            [effect]
-                apply_to=overlay
-                add="misc/arcane-icon.png"
-            [/effect]
-        [/object]
-    )}
+#define FIND_NEARBY_NOT_DACYN_PREFER_HEROES X Y LIMIT
+    {FIND_NEARBY id=Gweddry {X} {Y} {LIMIT}}
+    [if]
+        [not]
+            [variable]
+                name=found_unit.length
+                greater_than=0
+            [/variable]
+        [/not]
+
+        [then]
+            {FIND_NEARBY id=Owaec {X} {Y} {LIMIT}}
+            [if]
+                [not]
+                    [variable]
+                        name=found_unit.length
+                        greater_than=0
+                    [/variable]
+                [/not]
+
+                [then]
+                    {FIND_NEARBY id=Terraent {X} {Y} {LIMIT}}
+                    [if]
+                        [not]
+                            [variable]
+                                name=found_unit.length
+                                greater_than=0
+                            [/variable]
+                        [/not]
+
+                        [then]
+                            {FIND_NEARBY id=Konrad {X} {Y} {LIMIT}}
+                            [if]
+                                [not]
+                                    [variable]
+                                        name=found_unit.length
+                                        greater_than=0
+                                    [/variable]
+                                [/not]
+
+                                [then]
+                                    {FIND_NEARBY (
+                                        side=1
+                                        [not]
+                                            id=Dacyn
+                                        [/not]
+                                    ) {X} {Y} {LIMIT}}
+                                    [if]
+                                        [not]
+                                            [variable]
+                                                name=found_unit.length
+                                                greater_than=0
+                                            [/variable]
+                                        [/not]
+
+                                        [then]
+                                            {FIND_NEARBY id=Dacyn {X} {Y} {LIMIT}}
+                                        [/then]
+                                    [/if]
+                                [/then]
+                            [/if]
+                        [/then]
+                    [/if]
+                [/then]
+            [/if]
+        [/then]
+    [/if]
+
+#enddef
+
+# copied (with modifications) from https://wiki.wesnoth.org/WML_Utilities
+#define FIND_NEARBY FLT X Y LIMIT
```

Comment:
Probably a bit late to mention this, but… doesn't Wesnoth Lua Pack also have a tag with a similar effect? Maybe it's not quite the same…